crates/topos-tce-api/src/grpc/mod.rs

```diff
@@ -213,58 +215,38 @@ impl ApiService for TceGrpcService {
 
         let subnet_ids = data.subnet_ids;
 
-        let (sender, receiver) = oneshot::channel();
-
         let subnet_ids: Vec<SubnetId> = subnet_ids
             .into_iter()
             .map(TryInto::try_into)
             .map(|v| v.map_err(|e| Status::internal(format!("Invalid subnet id: {e}"))))
             .collect::<Result<_, _>>()?;
 
-        if self
-            .command_sender
-            .send(InternalRuntimeCommand::GetLastPendingCertificates { subnet_ids, sender })
-            .await
-            .is_err()
-        {
-            return Err(Status::internal(
-                "Can't get last pending certificates: sender dropped into channel",
-            ));
-        }
-
-        receiver
-            .map(|value| match value {
-                Ok(Ok(map)) => Ok(Response::new(GetLastPendingCertificatesResponse {
-                    last_pending_certificate: map
-                        .into_iter()
-                        .map(|(subnet_id, last_pending_certificate)| {
-                            (
-                                base64::engine::general_purpose::STANDARD
-                                    .encode(subnet_id.as_array()),
-                                {
-                                    LastPendingCertificate {
-                                        index: last_pending_certificate
-                                            .as_ref()
-                                            .map(|(_cert, index)| *index)
-                                            .unwrap_or_default()
-                                            as i32,
-                                        value: last_pending_certificate
-                                            .map(|(cert, _index)| cert)
-                                            .map(Into::into),
-                                    }
-                                },
-                            )
-                        })
-                        .collect(),
-                })),
-                Ok(Err(e)) => Err(Status::internal(format!(
-                    "Can't get last pending certificates: {e}"
-                ))),
-                Err(e) => Err(Status::internal(format!(
-                    "Can't get last pending certificates: {e}"
-                ))),
+        let last_pending_certificate = self
+            .store
+            .get_pending_certificates_for_subnets(&subnet_ids)
+            .map_err(|e| Status::internal(format!("Can't get last pending certificates: {e}")))?
+            .into_iter()
+            .map(|(subnet_id, (index, maybe_certificate))| {
+                (
+                    base64::engine::general_purpose::STANDARD.encode(subnet_id.as_array()),
```

Comment:
So this line looked annoyingly long and I went to check the docs to see if there's not something less fugly, but no, this is how it's supposed to be used (!). I did some more digging and it seems like the `base64` maintainer is struggling to keep up maintenance. One notable feature that is missing is constant time encoding/decoding. 

What do you think about switching to the RustCrypto `base64ct` crate instead? I don't think being constant time matters in this particular code, but if we have a single crate in the code base that happens to be constant time then we save our future selves a papercut: no need to even think about "so do I need the constant time base64 crate here or is the other one ok?".

Reply:
Handled by another PR